feat: add Prometheus metrics tracking for http requests

as-flow opened the pull request:
Refs: AB-487

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Added a Prometheus metrics endpoint for monitoring.
  * Tracks HTTP requests (count and duration) and exposes metrics including method, normalized endpoint, status, organization, application, and instance type.
* **Chores**
  * Added prometheus-client dependency to enable metrics collection.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

as-flow committed:
feat: add Prometheus metrics tracking for http requests

Refs: AB-487

as-flow committed:
chore: remove reliance on private method

UladzislauK-Writer commented on src/writer/serve.py:
```diff
@@ -159,6 +160,43 @@ async def lifespan(asgi_app: FastAPI):
+            return response
+        
+        method = request.method
+        endpoint = request.url.path
+        
+        # Skip static assets
+        if (
+            endpoint.startswith("/static/") or
+            endpoint.startswith("/assets/") or
+            endpoint.endswith((".ico", ".png", ".jpg", ".jpeg", ".gif", ".svg", ".css", ".js", ".woff", ".woff2"))
+        ):
+            response = await call_next(request)
+            return response
+        
+        try:
+            response = await call_next(request)
+            status_code = response.status_code
+        except Exception:
+            status_code = 500
+            raise
+        finally:
+            duration = time.time() - start_time
+            # Get instance metadata from environment variables
+            org_id = os.getenv("WRITER_ORG_ID", "unknown")
+            app_id = os.getenv("WRITER_APP_ID", "unknown")
+            instance_type = "editor" if serve_mode == "edit" else "agent"
+            
+            track_http_request(method, endpoint, status_code, duration, org_id, app_id, instance_type)
+        
+        return response
```
@as-flow  in general, instead of "blacklisting" routes that we don't want to track I think it's better to  "whitelist" ones that we do want. This way we won't have anything unexpected coming in even if somebody unaware of metrics decides to add a new route that would explode the cardinality

UladzislauK-Writer commented on src/writer/serve.py:
```diff
@@ -223,6 +261,11 @@ def _get_edit_starter_pack(payload: InitSessionResponsePayload):
+    def metrics():
+        """Prometheus metrics endpoint."""
+        return Response(get_metrics(), media_type=get_metrics_content_type())
+    
```
Lets be consistent and keep it under the `/api` route same as everything else

as-flow committed:
feat: enhance Prometheus metrics tracking by normalizing endpoint paths and adding endpoint filtering

- Introduced `normalize_endpoint` function to reduce metric cardinality by replacing dynamic segments with placeholders.
- Added `should_track_endpoint` function to filter tracked endpoints based on a predefined whitelist.
- Updated HTTP request tracking to utilize normalized endpoints.
- Changed metrics endpoint from `/metrics` to `/api/metrics` for consistency.
